### ISSUE TITLE
`ctrl-f` `ctrl-x` `A` do not accept autosuggestions in vi mode anymore

### ISSUE THREAD

**Issue by despairblue** [fish-shell/fish-shell#3541]
- [x] Have you checked if problem occurs with [fish 2.4.0](/fish-shell/fish-shell/releases/tag/2.4.0)?
- [x] Tried fish without third-party customizations *(check `sh -c 'env HOME=$(mktemp -d) fish'`)*?

**fish version installed** *(`fish --version`)*:
2.4.0

**OS/terminal used**: 
Mac OS Terminal
Hyperterm

Talk about the the issue here. 

## Reproduction steps
1. start a clean fish shell (`sh -c 'env HOME=$(mktemp -d) fish`)
2. `fish_vi_key_bindings`
3. `echo $PATH`
4. type `echo` and try to accept the autosuggestion with `ctrl-f` or `ctrl-x` (using right arrow works)

Also using `A` in command mode does not accept it anymore.

## Results
Autosuggestion is not accepted.

**Comment by faho** [comment]
This is as expected. The vi-bindings now behave vi-ish and only share select bindings with the default emacs-bindings.

To undo these specific ones, use

```fish
function fish_user_key_bindings
    for mode in insert default visual
        bind -M $mode \cf forward-char
    end
end
```

**Comment by jwoudenberg** [comment]
I just noticed this as well. Is there a default key shortcut to accept auto-suggestions in vim mode? From the documentation I get the idea Ctrl-x should do the trick, but that doesn't seem to work either.

**Comment by faho** [comment]
>Is there a default key shortcut to accept auto-suggestions in vim mode?

Right-arrow and end in insert mode, a lot of stuff in normal mode. Basically anything that moves the cursor over the suggestion will accept that part of it, except for "forward-char" which accepts the entire thing.

**Comment by faho** [comment]
> From the documentation I get the idea Ctrl-x should do the trick

That's an oversight in the docs, btw.

**Comment by jwoudenberg** [comment]
Thanks for your response! I set up ctrl-f like it used to work before and it's working great.

Perhaps having some shortcut bound to a Ctrl key combination by default would be nice though, as I don't think many will consider it very vi-ish to reach for the end or arrow keys :smile:. Perhaps Ctrl-o is an idea, which in vim triggers omni-completion.

**Comment by despairblue** [comment]
@faho Thanks for the clarification.

> Basically anything that moves the cursor over the suggestion will accept that part of it, except for "forward-char" which accepts the entire thing.

Does that mean that `A` should accept it?

**Comment by faho** [comment]
>Does that mean that A should accept it?

Well.... no. 

This is related to issue #3299. The problem is that we want to accomodate both

- "I'd like to edit what I typed!"

and

- "I'd like to get this awesome suggestion!"

so end-of-line only moves _up_ to the suggestion if it's not there. Executing it again will accept it. Now, A in normal/default-mode is bound to "end-of-line force-repaint" and switches to insert mode (obviously). So it would accept the suggestion if you used it right in front of it. Only you can't, because escape isn't _just_ bound to switching mode (or cancel or something), it also does "backward-char" because that's what vi does (for a reason I never understood). So it's impossible to use "A" to accept the suggestion, even though the binding actually should. You would currently need to redefine it or escape to do it.

I'd be open to suggestions in #3299.

**Comment by krader1961** [comment]
Also, for everyone that wants the legacy hybrid bindings that should be trivial after issue #3558 is resolved. You can test the fix in PR #3558.

**Comment by krader1961** [comment]
Closing since there have been no updates in over a month. Feel free to reopen if you have further questions or open a new issue if you feel the behavior can be improved.
